docs/DeveloperGuide.md

```diff
@@ -1,37 +1,1036 @@
+`Workout` object:
+1. Name of the exercise
+2. Number of repetitions associated with the exercise in (1).
+
+Note that these methods are not shown in the sequence diagram to improve the readability of the sequence diagram.
+
+(Steps 7.1 to 7.6) Next, validity checks of the user input are carried out to ensure that the data entered is valid as a
+new `Workout` object. The requirements for a valid new `Workout` object is as follows:
+- [x] The exercise name must exist in `ExerciseList`'s `exerciseList`, which is an `ArrayList<String>` of exercise 
+names.
+- [x] The repetition value must be a non-negative integer greater than 0.
+- [x] The (exercise name, repetition value) must not already exist in the list of workouts maintained in
+`WorkoutList`. For example, if a workout of 20 reps of push-ups is already stored in the list,
+it cannot be created again.
+
+If any of the three requirements are not met, either an `InvalidExerciseException` or an
+`InvalidWorkoutException` is thrown, and the entire workout creation process is aborted.
+
+Again, note that the actual method calls to check for the validity of the user input are not shown in the sequence
+diagram for the same reason as mentioned above.
+
+![creatAndAddWorkout() Sequence Diagram (Part 2)](uml/sequenceDiagrams/images/CreateAndAddWorkout-Part2.png)
+
+If the above checks pass, Step 7.7 will create the new `Workout` object with the user-specified exercise name and
+repetition value. Once that is done, a key of the `Workout` object will be generated in Step 7.9 (see the 
+[Design Considerations](#design-considerations-for-creating-a-new-workout) section for more details of the `HashMap`
+implementation), before storing the key-`Workout` pair in `workoutsHashMapList` which in turn is stored in `WorkoutList` 
+in Step 7.11. In Step 7.13, the key of the newly-created `Workout` object is added to the `workoutsDisplayList`, an 
+`ArrayList<String>` object stored in `WorkoutList`. This ArrayList will be used for displaying the workouts when the 
+command `workout /list` is entered by the user. This is the final step of `WorkoutList#createAndAddWorkout()`.
+
+(Step 9) Upon returning to `WorkoutCommand`, `UI#printNewWorkoutCreatedMessage()` is called to display a response to
+the user via the terminal. The following is an example of a response after the user entered `workout /new russian twist 
+/reps 50`:
+```
+----------------------------------------------------------------------
+Alright, the following workout has been created:
+
+	russian twist (50 reps)
+
+----------------------------------------------------------------------
+```
+
+(Step 12) `FileManager#writeNewWorkoutToFile` is called to write the newly-created `Workout` object's data into 
+`workouts.txt` which is stored on the user's local filesystem.
+
+This completes the process of adding a new workout to WerkIt!
+
+##### Design Considerations for Creating a New Workout
+###### HashMaps - Motivation
```

Comment:
Are these headers not too small? They are smaller than the normal text.

![image](https://user-images.githubusercontent.com/69353464/160966542-e4a269b8-fbe4-4949-ab42-b0329627a6d3.png)